knotx-example/knotx-example-monolith/src/test/java/com/cognifide/knotx/monolith/SampleApplicationTest.java

```diff
@@ -37,7 +37,6 @@
 import io.vertx.ext.unit.TestContext;
```

Comment:
I'm missing following tests - potentially integration. Since the headers are crucial, we'd need to have assurance on test level - following scenarios should be used:

### Passing headers to repository
- Original request having headers (Host + some other header)
- Http Repository is called by HTTP with the same headers as in original request

### Passing response headers + status code from repository
- Assuming all headers from services are filtered-out (or just template without any hbs snippets - to avoid service calls)
- Repository respond with few headers, e.g. (Location, Cache-Control, X-Robots-Tag)
- Repository respond with 200 and 301 (second scenario)
- Client gets repository statusCode + repository headers

### Passing all original request headers to service calls
- No headers filtered
- Original request having headers (Host + some few other headers)
- Service call to target service contains all headers that were in the request

### Passing allowed request headers to service calls
- configured filtering out one or more header
- Original request having headers (Host + some few other headers)
- Service call to target service contains only headers that are allowed.